File: bindings/ruby/test/test_render.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

$LOAD_PATH.unshift File.expand_path('../lib', __dir__)
require 'spannerplan'

repo_root = File.expand_path('../../..', __dir__)
fixture = File.join(repo_root, 'testdata/reference/dca.yaml')
golden = File.join(repo_root, 'testdata/golden/dca_plan_current.txt')

plan = File.read(fixture)
output = Spannerplan.render_tree_table_json(plan, mode: 'PLAN', format: 'CURRENT')
expected = File.read(golden)

unless output == expected
  warn 'output does not match golden dca_plan_current.txt'
  exit 1
end

unless output.include?('Distributed Cross Apply')
  warn 'expected marker not found in rendered output'
  exit 1
end

begin
  Spannerplan.render_tree_table_json('not json')
  warn 'expected RenderError for invalid json'
  exit 1
rescue Spannerplan::RenderError
  # expected
end

puts 'ok'

File: bindings/php/test_render.php
#!/usr/bin/env php
<?php

declare(strict_types=1);

require __DIR__ . '/src/Spannerplan.php';

$repoRoot = dirname(__DIR__, 2);
$fixture = $repoRoot . '/testdata/reference/dca.yaml';
$goldenPath = $repoRoot . '/testdata/golden/dca_plan_current.txt';

$plan = file_get_contents($fixture);
if ($plan === false) {
    fwrite(STDERR, "failed to read fixture: {$fixture}\n");
    exit(1);
}

$golden = file_get_contents($goldenPath);
if ($golden === false) {
    fwrite(STDERR, "failed to read golden: {$goldenPath}\n");
    exit(1);
}

$sp = new Spannerplan();
$output = $sp->renderTreeTableJson($plan, 'PLAN', 'CURRENT');

if ($output !== $golden) {
    fwrite(STDERR, "output does not match golden dca_plan_current.txt\n");
    exit(1);
}

if (strpos($output, 'Distributed Cross Apply') === false) {
    fwrite(STDERR, "expected marker not found in rendered output\n");
    exit(1);
}

echo "ok\n";
